File: Java/1299. Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int n = arr.length;
        int m = arr[n-1];
        arr[n-1] = -1;
        
        for(int i=n-2; i>=0; i--){
            int temp = arr[i];
            arr[i] = m;
            m = Math.max(m, temp);
        }
        
        return arr;
    }
}

File: Python3/1299. Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        m = arr[n-1]
        arr[n-1] = -1
        
        for i in range(n-2, -1, -1):
            arr[i], m = m, max(m, arr[i])
            
        return arr
                